js/query.js

```diff
@@ -1,70 +1,129 @@
+        }
+
+        fetch(queryUrl)
+            .then(
+                function(response) {
+                    if (response.status !== 200) {
+                        console.log('Looks like there was a problem. Status Code: ' +
+                            response.status);
+                        return;
+                    }
 
-            fetch(queryUrl)
-                .then(
-                    function(response) {
-                        if (response.status !== 200) {
-                            console.log('Looks like there was a problem. Status Code: ' +
-                                response.status);
-                            return;
+                    response.json().then(function(data) {
+                        if (data.total !== 0 && data.success !== false) {
+                            topHit = data.hits[0]
+                            var basics = { 'geneSymbol': topHit.symbol, 'geneName': topHit.name, 'geneId': {'db': 'https://www.ncbi.nlm.nih.gov/gene/', 'ident': topHit._id}, 'matchScore': topHit._score, 'hits': data.hits.length };
+                            displayData(basics)
+                            annotateGene(topHit._id);
+                        } else {
+                            var empty = { 'hits': 'No hits', 'matchScore': '0' };
+                            displayData(empty)
+                            hideData(document.getElementById('infoDiv'));
+                            hideData(document.getElementById('locDiv'));
+                            hideData(document.getElementById('summaryDiv'));
                         }
+                    });
+                }
+            )
+            .catch(function(err) {
+                console.error('Fetch Query Error', err);
+            });
+    }
 
-                        response.json().then(function(data) {
-                            console.log(data);
-                            if (data.total !== 0 && data.success !== false) {
-                                top_hit = data.hits[0]
-                                var symbol = top_hit.symbol
-                                var geneId = top_hit._id
-                                var geneName = top_hit.name
-                                document.getElementById('basics').textContent = 'Official Gene Symbol: ' + symbol + '; Official Gene Name: ' + geneName + '; Entrez ID: ' + geneId;
-                                annotateGene(geneId);
-                            } else {
-                                document.getElementById('basics').textContent = 'No hits.'
-                                document.getElementById('summary').textContent = ''
-                            }
-                        });
-                    }
-                )
-                .catch(function(err) {
-                    console.error('Fetch Query Error', err);
-                });
+    function displayData(dataArray) {
+        for (data in dataArray) {
```

Comment:
Don't use `for...in` loops to loop through arrays in JavaScript. [Read here.](https://stackoverflow.com/a/500878)

Use a simple for loop instead.

Reply:
Python background flaring. Use case for `for...in` loops is unexpected.

Reply:
Actually, after reading/testing, my use case is appropriate for this. The `dataArray` is actually an Object, not an array. Really just a poorly named variable.